source/tyr/tyr/poi_to_excluded_zones.py

```diff
@@ -0,0 +1,68 @@
+import argparse
+import csv
+import json
+import logging
+
+
+def poi_to_excluded_zones(poi_file, output_dir, instance_name):
+    tmp_path = "tmp/poi_{}".format(instance_name)
+    import zipfile
+
+    with zipfile.ZipFile(poi_file, 'r') as zip_ref:
+        zip_ref.extractall(tmp_path)
+
+    excluded_zones = {}
+    excluded_geometries_ids = {}
+
+    # get excluded zones
+    with open(tmp_path + "/poi_properties.txt") as csvfile:
+        reader = csv.reader(csvfile, delimiter=';', quotechar='"')
+        for row in reader:
+            if row[1].lower() != "excluded_zones":
+                continue
+            excluded_zones[row[0]] = json.loads(row[2])
+
+    # find geometry id
+    with open(tmp_path + "/poi.txt") as csvfile:
+        reader = csv.reader(csvfile, delimiter=';', quotechar='"')
+        for row in reader:
+            if row[0] not in excluded_zones:
+                continue
+            excluded_geometries_ids[row[0]] = row[7]
+
+    if excluded_geometries_ids.keys() != excluded_zones.keys():
+        logger.error("not all excluded zone's pois are found in poi.txt")
+        logger.error("excluded_geometries_ids: {}".format(excluded_geometries_ids.keys()))
+        logger.error("excluded_zones: {}".format(excluded_zones.keys()))
+
+    # read geometries
+    geometries_shapes = {}
+    with open(tmp_path + "/geometries.txt") as csvfile:
+        reader = csv.reader(csvfile, delimiter=';', quotechar='"')
+        for row in reader:
+            geometries_shapes[row[0]] = row[1]
+
+    for poi_id, zones in excluded_zones.items():
+        geometry_id = excluded_geometries_ids.get(poi_id)
+        if not geometry_id:
+            logger.error("{} could not be found in poi.txt".format(row[0]))
+        shape = geometries_shapes.get(geometry_id)
+        if not shape:
+            logger.error("{} could not be found in geometries.txt".format(geometry_id))
+
+        for i, zone in enumerate(zones):
+            output_id = "{}_{}_{}".format(poi_id, i, instance_name)
+            output = {'id': output_id}
+            output.update(zone)
```

Comment:
you can have a poi with several zones but all with the same shape ? :thinking:

Reply:
yeahhhhh, I admit that this was not well thought through... 

the need is for a given shape, we wish it could be excluded based on a list of modes and a list of ranges of date, so the ideal format in my opinion is

```json
{
  "name": "poi:toto",
  "shape": "POLYGON ((...))",
  "excluded_on": [
    {
      "modes": [
        "walking",
        "bike"
      ],
      "periods": [
        {
          "from": "20240708",
          "to": "20240709"
        },
        {
          "from": "202407010",
          "to": "20240711"
        }
      ]
    },
    {
      "modes": [
        "car"
      ],
      "periods": [
        {
          "from": "202407010",
          "to": "202407011"
        }
      ]
    }
  ]
}
```